### PR TITLE
Scala3doc: improve stdlib loading

### PR DESCRIPTION
**Pull request by abgruszecki**
This PR fixes the issue that prevented us from loading stdlib together w/ other artifacts, and makes the doctool aware of stdlib patching going on in the compiler.

### CODE REVIEW

**Comment by abgruszecki** [comment]
Alright, there's one problem with this PR: it now documents multiple projects together, and those projects actually use different docstring syntax. See : https://scala3doc.virtuslab.com/pr-improve-stdlib-loading/scala3/-scala%203/-a-p-i/scala/-predef.html?query=object%20Predef%20extends%20LowPriorityImplicits .

**Comment by abgruszecki** [comment]
We do get the new definitions documented, though: https://scala3doc.virtuslab.com/pr-improve-stdlib-loading/scala3/-scala%203/-a-p-i/scala/-predef/summon.html

**Comment by abgruszecki** [comment]
> Alright, there's one problem with this PR: it now documents multiple projects together, and those projects actually use different docstring syntax. See : https://scala3doc.virtuslab.com/pr-improve-stdlib-loading/scala3/-scala%203/-a-p-i/scala/-predef.html?query=object%20Predef%20extends%20LowPriorityImplicits .

The best fix I can think of is to set default syntax to wiki and add `@syntax markdown` to every comment outside of stdlib. @romanowski if you have a better idea, let me know; this'll be a problem when trying to document multiple projects as well, but I can't think of any good solution.

File: scala3doc/src/dotty/dokka/tasty/ClassLikeSupport.scala [hunk 1]
```diff
@@ -203,6 +205,31 @@ trait ClassLikeSupport:
         inherited.flatMap(s => parseInheritedMember(s))
     }
 
+    /** Extracts members while taking Dotty logic for patching the stdlib into account. */
+    def extractPatchedMembers: Seq[Member] = {
+      val ownMembers = c.extractMembers
+      def extractPatchMembers(sym: Symbol) = {
```
Comment (romanowski) [inline scala3doc/src/dotty/dokka/tasty/ClassLikeSupport.scala:211]: Shouldn't that logic be placed into TASY? I mean  people may wan to parse tasty from stdlib and they will be really supprised what they will get

**Review by romanowski — CHANGES_REQUESTED** [review]
Generally I think we should try to not hardcode any stdlib related logic in doctool.

**Comment by romanowski** [comment]
I think we can use similat approach as for sourcelinks or we will use for extenrnal docs. So we can either pass a single value (as we do now) or pass pairs `filePattern,wiki syntax` since people may have similar problems in thier repos (e.g. when documenting shared code and new pieces for scala 3)

File: scala3doc/src/dotty/dokka/tasty/ClassLikeSupport.scala (hunk 1 above)
Comment (abgruszecki) [reply scala3doc/src/dotty/dokka/tasty/ClassLikeSupport.scala:211]: Tasty Reflect is already aware of how the compiler patches stdlib definitions. Our problem is that we rely on class bodies to document classes, and those aren't patched.

We can open a separate issue for patching class bodies as well.

File: scala3doc/src/dotty/dokka/tasty/TastyParser.scala [hunk 2]
```diff
@@ -30,17 +30,22 @@ case class DokkaTastyInspector(parser: Parser)(using ctx: DocContext) extends Do
...
-    def driFor(link: String): Option[DRI] =
-      val symOps = new SymOps[q.type](q)
-      import symOps._
-      Try(QueryParser(link).readQuery()).toOption.flatMap(q =>
-        MemberLookup.lookupOpt(q, None).map{ case (sym, _) => sym.dri}
-      )
+    // NOTE we avoid documenting definitions in the magical stdLibPatches directory;
+    // the symbols there are "patched" through dark Dotty magic onto other stdlib
+    // definitions, so if we documented their origin, we'd get defs with duplicate DRIs
+    if !root.symbol.show.startsWith("scala.runtime.stdLibPatches") then
```
[the remark this answers is not in the document]
Comment (abgruszecki) [reply scala3doc/src/dotty/dokka/tasty/TastyParser.scala:36]: Can we have a separate issue for that? I can already see a bunch of things that would need to be considered with such an option (should the option take a list of full package names? a list of FQN string prefixes?) and I'd rather discuss them separately.

**Comment by abgruszecki** [comment]
> I think we can use similat approach as for sourcelinks or we will use for extenrnal docs. So we can either pass a single value (as we do now) or pass pairs `filePattern,wiki syntax` since people may have similar problems in thier repos (e.g. when documenting shared code and new pieces for scala 3)

I agree, but the filePattern should probably reference the path to the `.tasty` file we loaded the definition from, otherwise it will be painful to use. I looked into it and we can't get that right now.

Right now I'm looking into just defaulting to wiki syntax for stdlib docs. That may require the least changes to already-existing docstrings.

**Review by romanowski — APPROVED** [review]

**Comment by romanowski** [comment]
Lets merge it for now and remove hardcoded bits in follow up PRs

**Pull request merged**